docs2/site/docs/migrations/migration5.md

```diff
@@ -602,12 +602,13 @@ There are a number of other minor issues fixed; see these links for more details
 - https://github.com/graphql-dotnet/graphql-dotnet/issues/3002
 - https://github.com/graphql-dotnet/graphql-dotnet/pull/3004
 
-### 20. `Authorize` and `AuthorizeWithRoles` extension methods added in GraphQL 5.1.0 and 5.1.1
+### 20. `Authorize`, `AuthorizeWithRoles` and `AllowAnonymous` extension methods added in GraphQL 5.1.0 and 5.1.1
 
-This allows for specifying roles rather than just policies that can be used to validate a request.
+`AuthorizeWithRoles` allows for specifying roles rather than just policies that can be used to validate a request.
 `Authorize` can be used to specify that only authentication is required, without specifying any specific roles or policies.
-As with `AuthorizeWithPolicy` (renamed from `AuthorizeWith`), it requires support by a third-party
-library to perform the validation.
+`AllowAnonymous` indicates that anonymous access should be allowed to a field of a graph type requiring authorization,
```

Comment:
I suggest not to narrow meaning of allowanonymous only to fields like for actions in ASP.NET controllers. It's GraphQL world.

Reply:
In this case, I'm not really thinking about how it relates to ASP.NET (apart from the naming and general concept), but rather the practical usage of this attribute in the GraphQL world.  I'm just not sure how AllowAnonymous would be useful to apply to a graph, since graphs are generally abstract from other graphs.  For instance, if Product is marked as AllowAnonymous, and if MasterProductsTable is protected, querying for all products within MasterProductsTable should still be refused.  The same thought process applies to input object graphs and scalars, leaving only fields at this time.

Within ASP.NET Core, only one action can ever execute, so AllowAnonymous simply overrides all Authorize attributes that are parent to it in the chain.  In this respect, my suggestion is unlike ASP.NET Core.  It's simply what works best for GraphQL.

I can remove the attribute usage restriction, but even if we did so, I figured that the xml comments should explain anticipated usage of the attributes, even knowing that implementation is not part of this library.  If nothing else, use the word "typically" and retain the wording/example.

Do you suggest I remove the attribute usage restrictions?  Do you have a suggestion of rewording here?

Reply:
> implementation is not part of this library

This is true and we must leave some space for maneuver.

> word "typically" and retain the wording/example

Yes, let's insert _typically_ somewhere. The general meaning will be the same, but the phrase will sound more accurate.

Reply:
```suggestion
`AllowAnonymous` typically indicates that anonymous access should be allowed to a field of a graph type requiring authorization,
```

Reply:
See above (merged).